File: AXServer/Sources/CodeAlpha/websocket/messages/models/XCodeFocusStatusChange.swift
import Foundation

enum XCodeFocusElement: String, Codable {
  case Editor
  case App
}

struct XCodeFocusStatusChange: Codable {
  let focusElementChange: XCodeFocusElement
  let isInFocus: Bool
  let uiElementX: Float
  let uiElementY: Float
  let uiElementW: Float
  let uiElementH: Float
}

File: widget-prototype/src-tauri/src/websocket/accessibility_messages/models/x_code_focus_status_change.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
pub enum XCodeFocusElement {
    Editor,
    App,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct XCodeFocusStatusChange {
    pub focus_element_change: XCodeFocusElement,
    pub is_in_focus: bool,
    pub ui_element_x: f64,
    pub ui_element_y: f64,
    pub ui_element_w: f64,
    pub ui_element_h: f64,
}
